backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -218,17 +226,20 @@ def sanitize_records(self, records, selected_trees, filters):
                 archSummary["status"][status] += 1
 
             if build_id not in processed_builds:
-                processed_builds.add(build_id)
                 build = get_build(r, current_tree["index"])
+                processed_builds.add(build_id)
 
                 if len(filters_map['build']) == 0 or self.pass_in_filters(build, filters_map['build']):
                     builds["items"].append(build)
+                    build_is_valid = build["valid"]
```

Comment:
You are using this variable outside of this if scope, which may lead to a `NameError`, IMO we don't need to create this var and just use build["valid"] since you are using it only once

Reply:
From my understanding build["valid"] could also be outside of scope (since it's created inside an if) and even if not (I could move it to be outside of the if), build is not always added to the builds["items"] list, that's why I'm defining this variable inside the if statement that appends the build to build["items"] (if the build is not part of the final list, then I believe it shouldn't count as a failed build with unknown issue even if the others constraints apply). Is my reasoning correct? If so, I think that defining `build_is_valid = None` before both if statements would work

Reply:
you are right, it could be `record["build__valid"]`